src/qml/MapCanvas.qml

```diff
@@ -266,6 +266,50 @@ Item {
         }
     }
 
+    DragHandler {
+        target: null
+        enabled: interactive
+        acceptedDevices: PointerDevice.Stylus | PointerDevice.Mouse
+        grabPermissions: PointerHandler.TakeOverForbidden
+        acceptedButtons: Qt.MiddleButton
+
+        property real oldTranslationY: 0
+        property bool translationThresholdReached: false
+
+        onActiveChanged: {
+            if (active)
+            {
+                freeze('rotate')
+                oldTranslationY = 0
+                translationThresholdReached = false
+            }
+            else
+            {
+                unfreeze('rotate')
+            }
+        }
+
+        onTranslationChanged: {
+            if (active)
+            {
+                if (translationThresholdReached)
+                {
+                    if (oldTranslationY != 0)
+                    {
+                        mapCanvasWrapper.rotate(oldTranslationY - translation.y)
+                    }
+                    oldTranslationY = translation.y
+                    translationThresholdReached = true
+                }
+                else if (Math.abs(oldTranslationY - translation.y) > 30)
```

Comment:
I think that could be smaller.
Also possibly we could rotate relative to the original finger position (not position when threshold reached)

Reply:
@m-kuhn , did you give the APK a try? I also felt like it could go down to maybe 20 degrees.

Re rotating relative to the original finger position, I tried it at the beginning and it felt "wrong/broken" to me. It's also not in line with what other mapping apps are doing (e.g. Google Maps).